test/run_test.sh

```diff
@@ -89,6 +89,21 @@ if [[ "$OSTYPE" != "msys" ]]; then
 
     distributed_tear_down() {
       rm -rf "$TEMP_DIR"
+      unset WORLD_SIZE
```

Comment:
@apaszke WORLD_SIZE is cleaned here. Hmmm somehow it doesn't show up in the commit you reviewed.